src/util.js

```diff
@@ -57,8 +57,32 @@
     return false;
   }
 
+
+  /*
+   * Get iterator of object which returns a { done: false, value: expression }
+   *   if it has not reached the end of the object, or otherwise it returns
+   *   { done: true, value: undefined }.
+   * @param {!Object} object: The object to iterate over.
+   * @param {Function} callback: A function to be called by using key and value
+   *   as parameters and will return the value in a defined format
+   */
+  function iterator(object, callback) {
+    if (!(this instanceof arguments.callee))
+      return new arguments.callee(object, callback);
+    var index = 0;
+
+    this.next = function next() {
+      if (index < object.length) {
+          var key = index;
+          var value = object[index++];
+          return { done: false, value: callback(key, value) };
+      } else return { done: true, value: undefined };
+    };
```

Comment:
I think this can be done more clearly by just using an object:

```javascript
function iterator(object, callback) {
  return {
    index: 0,
    next: function() {
      if (index < object.length) {
        var key = this.index;
        var value = object[this.index++];
        return { done: false, value: callback(key, value) };
       }
       return { done: true, value: undefined };
    }
  }
}
```